drivers/gpu/drm/meson/meson_drv.c

```diff
@@ -1043,7 +1041,7 @@ static void update_scaler(struct drm_crtc *crtc)
 		/* Basic scaler config */
 		osd_reg_write(VPP_OSD_SC_CTRL0,
 			      (1 << 3) /* Enable scaler */ |
-			      (0 << 2) /* Select OSD1 */);
+			      (1 << 2) /* Select OSD1 */);
```

Comment:
I think the comment is incorrect -- probably my fault.

If you compare https://github.com/endlessm/linux-meson/blob/master/drivers/amlogic/display/osd/osd_hw.c#L1898 and https://github.com/endlessm/linux-meson/blob/master/drivers/amlogic/display/osd/osd_hw.c#L2093 you'll see that the OSD1 vs. OSD2 select is at bit position 0 (and the PDF agrees). Unfortunately, bit 2 doesn't seem to be labeled at all in the PDF I have, which seems to be the wrinkle.